ConsoleCalculator/Calculator.cs

```diff
@@ -4,11 +4,166 @@ namespace ConsoleCalculator
 {
     public class Calculator
     {
+        public string displayValue;
+        public char[] operators = { '.', '+', '-', '/', '*', 's', 'c', '=' };
+        public decimal currentOperand;
+        public char currentOperator;
+        public decimal result;
 
-        public string SendKeyPress(char key)
+    public string SendKeyPress(char key)
         {
-            // Add your implementation here.
-            throw new NotImplementedException();
+            if (Char.IsDigit(key))
+            {
+                DigitHandle(key);
+            }
+
+            if (IsOperator(key))
+            {
+                OperatorHandle(key);
+            }
+
+            return ShowDisplayValue();
+        }
+
+        public void DigitHandle(char key)
+        {
+            if (!string.IsNullOrEmpty(displayValue) && displayValue.Contains("."))
+            {
+                SetDisplayValue(displayValue + key);
+            }
+            else
+            {
+                currentOperand = decimal.Parse(currentOperand.ToString() + key);
+                SetDisplayValue(currentOperand.ToString());
+            }
+
+        }
+
+        public void OperatorHandle(char key)
+        {
+            switch (key)
+            {
+                case '+':
+                    SetDisplayValue("+");
+                    HandleArithmeticOperation(key);
+                    break;
+                case '-':
+                    SetDisplayValue("-");
+                    HandleArithmeticOperation(key);
+                    break;
+                case '/':
+                    SetDisplayValue("/");
+                    HandleArithmeticOperation(key);
+                    break;
+                case '*':
+                    SetDisplayValue("*");
+                    HandleArithmeticOperation(key);
+                    break;
+                case '=':
+                    HandleArithmeticOperation(key);
+                    break;
+                case 'c':
+                    Reset();
+                    break;
+                case 's':
+                    ToggleSign();
+                    break;
+                case '.':
+                    SetDisplayValue(displayValue + ".");
+                    break;
+                default:
+                    break;
+            }
+        }
+
+        public void HandleArithmeticOperation(char key)
+        {
+            if (currentOperator == '\0')
+            {
+                currentOperator = key;
+                result = currentOperand;
+                currentOperand = 0;
+            }
+            else
+            {
+                PerformsArithmeticOperation(key);
+            }
+        }
+
+        public void PerformsArithmeticOperation(char key)
+        {
+            switch (currentOperator)
+            {
+                case '+':
+                    if (currentOperand > 0)
```

Comment:
Add comments for every validation you are performing... it is difficult to understand why you have written certain conditions